Deprecation fix: File.exists? => File.exist

ronen opened the pull request:
Address deprecation warning:

     ...ruby/2.1.3/gems/addressable-2.3.6/lib/addressable/uri.rb:265: warning: File.exists? is a deprecated name, use File.exist? instead

iainbeeston commented:
Or try running the tests with "rspec --warnings"

Any chance of getting this merged?

iainbeeston commented:
@sporkmonger Is there anything we can do to get this merged? It makes it very difficult to use addressable with warnings enabled

iainbeeston commented:
@ronen Do you think you could add `RSpec.configure { |c| c.warnings = true }` to `spec_helper.rb` (in this pull request) so @sporkmonger can see what we mean?

iainbeeston commented:
(And maybe prevent future problems too)

aplatypus commented:
Hi ...

I have been looking for the 'reason' for this message:

*  deprecated. use instead jars/installer

I've been searching on here and the web I haven't found anything except this issue.  Is this connected some way?  What does the message mean?

It came in the context of installing a gem called **middleman-blog**.

```
**jruby -S gem install middleman-blog**
Fetching: addressable-2.3.7.gem (100%)
deprecated. use instead jars/installer
Successfully installed addressable-2.3.7
Fetching: middleman-blog-3.5.3.gem (100%)
Successfully installed middleman-blog-3.5.3
2 gems installed
```

I didn't have any *verbose* or *warning* switches set though.  I am using the JRuby 9.0 version

    JRuby 9.0.0.0.pre1 (2.2.0p0) 2015-01-20 d537cab 
    Java HotSpot(TM) 64-Bit Server VM 25.25-b02 on 1.8.0_25-b18 +jit [Windows 7-amd64]

Many thanks,

  Will.

iainbeeston commented:
@aplatypus I think that's an unrelated deprecation message.

ronen commented:
@iainbeeston wrote:
> @ronen Do you think you could add RSpec.configure { |c| c.warnings = true } to spec_helper.rb (in this pull request) so @sporkmonger can see what we mean?

Added it.  That also exposed some variable-set-but-not-used warnings in template_spec.rb, which I fixed and likewise added to to the PR.

BTW with warnings enabled, MRI 2.2.0p0 gives a bunch of warnings `possible reference to past scope` in the specs -- it seems that's a bug which will be fixed in future releases, see https://bugs.ruby-lang.org/issues/10661 and https://bugs.ruby-lang.org/projects/ruby-trunk/repository/revisions/49082

sporkmonger commented:
Hey all, sorry it took me so long to get around to merging this.

iainbeeston commented:
:tada: :smile:

ronen commented:
@sporkmonger cool, thanks.

FYI also MRI 2.2.1 indeed removes the `possible reference to past scope` warnings.
